LoyaltyProgram_Intern/src/main/java/com/pgs/soft/controller/UserController.java

```diff
@@ -64,10 +62,18 @@ public String fillProfile(@Valid @RequestBody UserProfileDTO userProfileDTO){
 		return "Profile filled.";
 	}
 	
+	@RequestMapping(value = "/register", method=RequestMethod.GET)
+	public String registerView(@ModelAttribute("userDTO") UserDTO userDTO){
+		return "register";
+	}
+	
 	@RequestMapping(value = "/register", method=RequestMethod.POST)
-	public String register(@Valid @RequestBody UserDTO userDTO){
+	public String register(@Valid @ModelAttribute("userDTO") UserDTO userDTO, BindingResult result){
+		if(result.hasErrors()){
```

Comment:
Pleas change it to !hasErrors condition and then save. After that return "register"